lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -1125,40 +1157,6 @@ def GetAllInstancesInfo(self, hvparams=None):
         data.append(info)
     return data
 
-  @staticmethod
-  def _GenerateDiskAioCacheParameters(disk_aio, disk_cache, dev_type):
-    """Generate appropriate aio/cache parameters for QEMU
-
-    @type disk_aio: string
-    @param disk_aio: the instance's AIO parameter
-    @type disk_cache: string
-    @param disk_cache: the instance's disk cache parameter
-    @type dev_type: string
-    @param dev_type: the disk type in use
-    @rtype: string
-    @return: parameter string suitable for QEMU drive parameters
-
-    """
-
-    if (dev_type in constants.DTS_EXT_MIRROR
-            and dev_type != constants.DT_RBD):
-      logging.warning("KVM: overriding disk_cache setting '%s' with 'none'"
-                      " to prevent shared storage corruption on migration",
-                      disk_cache)
-      disk_cache = constants.HT_CACHE_NONE
```

Comment:
I wonder if we no longer need to override the cache mode to `none` for shared storage?

Reply:
Good catch! I must have overlooked this while refactoring.

Reply:
I have re-introduced this as part of the `_GetCacheSettings()` helper in the latest push.